source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -54,6 +54,46 @@ void SingleServerRequest::onFailure() {
   callbacks_.onResponse(Utility::makeError(Response::get().UpstreamFailure));
 }
 
+bool SingleServerRequest::onMovedRedirection(Common::Redis::RespValue& value) {
+  if ((incoming_request_.get() == nullptr) || (value.type() != Common::Redis::RespType::Error)) {
+    return false;
+  }
+  std::vector<absl::string_view> err = StringUtil::splitToken(value.asString(), " ", false);
+  if (err.size() != 3) {
+    return false;
+  }
+
+  const std::string host_address = std::string(err[2]); // ip:port
+  handle_ = conn_pool_.redirectRequest(host_address, *incoming_request_, *this);
+  return (handle_ != nullptr);
+}
+
+bool SingleServerRequest::onAskRedirection(Common::Redis::RespValue& value) {
+  if ((incoming_request_.get() == nullptr) || (value.type() != Common::Redis::RespType::Error)) {
```

Comment:
Would it worth moving these shared checks into a separate function? They are only used twice so ???

Reply:
I'm not sure. Each request type has its own set of additional checks, so I'm not sure how much gain we would get with a call to an inline function with just a couple of checks. Once I delete alot of the duplicate code in this file (see next push), it'll be a lot cleaner in general.

Reply:
Ok I'll wait for that to come in

Reply:
I've transformed the code into a static lambda function, and reduced some of the cut-and-paste duplication (replaced by the call to the lambda function).